spec/handlers/sessions_create_spec.rb

```diff
@@ -356,6 +356,70 @@
             end
           end
         end
+
+        context "linked to no one" do
+          context "but user already has already authenticated with the same provider before" do
+            context "new_social user" do
+              let!(:user) { FactoryGirl.create(:new_social_user) }
+              let!(:authentication) { FactoryGirl.create(:authentication, user: user) }
+
+              it "returns a status output instead of raising an exception" do
+                result = lambda { |uid|
+                            described_class.handle(
+                              user_state: user_state,
+                              request: MockOmniauthRequest.new(authentication.provider, uid, {})
+                            )
+                          }
+
+                result.call(111)
+
+                same_provider_different_uid = result.call(222)
+                expect{same_provider_different_uid}.not_to raise_exception
+                expect(same_provider_different_uid.outputs[:status]).to eq :same_provider
+              end
+            end
+
+            context "temp user" do
+              let!(:user) { FactoryGirl.create(:temp_user) }
+              let!(:authentication) { FactoryGirl.create(:authentication, user: user) }
+
+              it "returns a status output instead of raising an exception" do
+                result = lambda { |uid|
+                            described_class.handle(
+                              user_state: user_state,
+                              request: MockOmniauthRequest.new(authentication.provider, uid, {})
+                            )
+                          }
+
+                result.call(111)
+
+                same_provider_different_uid = result.call(222)
+                expect{same_provider_different_uid}.not_to raise_exception
+                expect(same_provider_different_uid.outputs[:status]).to eq :same_provider
+              end
+            end
+
+            context "non-temp user" do
+              let!(:user) { FactoryGirl.create(:user) }
+              let!(:authentication) { FactoryGirl.create(:authentication, user: user) }
+
+              it "returns a status output instead of raising an exception" do
+                result = lambda { |uid|
+                            described_class.handle(
+                              user_state: user_state,
+                              request: MockOmniauthRequest.new(authentication.provider, uid, {})
+                            )
+                          }
+
+                result.call(111)
+
+                same_provider_different_uid = result.call(222)
+                expect{same_provider_different_uid}.not_to raise_exception
+                expect(same_provider_different_uid.outputs[:status]).to eq :same_provider
+              end
+            end
+          end
+        end
```

Comment:
may not need it, but in case you are testing for lev errors, note that there is a `expect(blah).to have_routine_error(:some_error_code)`.  See `rails_helper.rb`